stage0/src/sev.rs

```diff
@@ -28,14 +28,64 @@ use oak_sev_guest::{
 use spinning_top::Spinlock;
 use x86_64::{
     instructions::tlb,
-    registers::control::Cr3,
+    registers::{control::Cr3, model_specific::Msr},
     structures::paging::{
         frame::PhysFrameRange, page::AddressNotAligned, Page, PageSize, PageTable, PageTableFlags,
         PhysFrame, Size1GiB, Size2MiB, Size4KiB,
     },
     PhysAddr, VirtAddr,
 };
 
+/// The identifier for the MSR used to set the default caching mode.
+const IA32_MTRR_DEFTYPE: u32 = 0x2ff;
+
+/// The cache memory type used with MTRR.  We only use Write-Protect mode which the Linux kernel
+/// expects to be enabled by the firmware in order to enable SEV.
+pub enum MemoryType {
+    WP = 5, // Write-Protect.
+}
+
+/// Models fields of the IA32_MTRR_DefType MSR.
+struct MTRRDefType {
+    mtrr_enable: bool,               // Set this to enable MTRR.
+    fixed_range_enable: bool,        // Set to enable fixed-range support.
+    default_memory_type: MemoryType, // The default caching mode.
+}
+
+impl From<MTRRDefType> for u64 {
+    fn from(deftype: MTRRDefType) -> Self {
+        let mtrr_enable = if deftype.mtrr_enable { 1 } else { 0 };
```

Comment:
`deftype.mtrr_enable.into()` should suffice as `From<bool>` is implemented for u64.